store/v2/storage/sqlite/db.go

```diff
@@ -203,36 +195,30 @@ func (db *Database) Prune(version uint64) error {
 		t2.version <= ?
 	) AND store_key != ?;
 	`
-
-	if _, err = tx.Exec(pruneStmt, version, reservedStoreKey); err != nil {
+	if err := db.executeTx(pruneStmt, version, reservedStoreKey); err != nil {
 		return fmt.Errorf("failed to exec SQL statement: %w", err)
 	}
 
-	// prune removed store keys
+	// prune removed stores
 	pruneRemovedStoreKeysStmt := `DELETE FROM state_storage
-	WHERE store_key in (
-		SELECT value FROM state_storage
-		WHERE store_key = ? AND key = ? AND version <= ?
+	WHERE store_key IN (
```

Comment:
Would it make sense to limit the delete to `version <= ?` ?
I think about odd edge cases where store foo is renamed to bar and back?

Reply:
The when the delete does not contain a version condition as well, it will delete everything with a matching store key.
For example
```
height 10: add store foo + some data
height 20: delete store foo
height 30: add store foo + some data
```
When prune is is run on height 31, it would find the store marked for delete (at height 20) and deletes everything although it is in use again.

This opens another question, if the store is not pruned, would it contain dirty data?